### ISSUE TITLE
ExcelFilterStyle , what is the name of the onClick methods for the apply and cancel button ? onFilteringDone doesnt work here

### ISSUE THREAD

**Issue by corganfuzz** [IgniteUI/igniteui-angular#4248]
In the ExcelFilterStyle box , what is the name of both onClick methods for binding the apply and cancel button ? onFilteringDone doesnt work here. cant find it in the documentation

onFilteringDone only works for `filterMode="quickFilter" `

![button](https://user-images.githubusercontent.com/11259414/54088654-609d7800-432e-11e9-8261-80fc9ba5ef3d.PNG)

 * igniteui-angular version: 7.2.0
 * browser: Chrome

**Comment by sstoyanovIG** [comment]
@corganfuzz We are going to emit onFilteringDone when 'Apply' button is clicked.

**Comment by corganfuzz** [comment]
@sstoyanovIG Thanks. please consider binding with the Search box too.  With `filterMode="quickFilter"` I have a sibling component that changes according to what a user types in this box, like I said this already work with quickfilter

**Comment by sstoyanovIG** [comment]
@corganfuzz The search box is used to filter the items below it, like in excel's filtering. It doesn't filter the grid.

**Comment by corganfuzz** [comment]
Oh. Right, lol. my bad sorry about that

**Comment by corganfuzz** [comment]
Wait. In quicktilter the search box also filters the items below it, since it had the onFilterDone() attached , then this method allows the search box to pass the filtering results to other sinlings components. 

So, correct me if im wrong But wouldnt it make more sense to attach onFiltermode() to the search box and create method(s) to apply and cancel, no?

**Comment by bkulov** [comment]
Hi @corganfuzz,

Quick filtering is meant to filter the grid items on the fly - when you type in the edit box the grid content is filtered. The Excel Style Filtering is something different. The filter is applied to grid's content when you click 'Apply' button. Anything you do in the Excel Style Filtering dialog will take effect once you press the 'Apply' button.

In the quick filtering each time you type something in the edit box triggers grid filtering. The grid is filtered and the 'onFilteringDone' event is emitted.

In the Excel Style Filtering the grid is filtered when you press the 'Apply' button and the 'onFilteringDone' event should be emitted (which is not currently happening and will be fixed once this PR gets merged #4260 ). In that regards you shouldn't consider Excel Style Filtering as a quick filter which filters the grid immediately. In order to detect when the grid is filtered you should subscribe to 'onFilteringDone' event. This is applicable to both filtering modes.

**Comment by corganfuzz** [comment]
I see, it’s the old “apply button” vs “data binding” conundrun. We run into it once in our api design. I guess they do achieve the same thing at the end. Just wish there was an onchange method on that search method, tho. ng-deep maybe ?😀

**Comment by bkulov** [comment]
I'm not sure I understand the question. Do you want to have an event emitted when you type in the search text box (shown on the image bellow)? What's the benefit of having such event? This search box only filters the list of items in the list bellow it so the user could find the item he wants to check without scrolling through the whole list. Typing in this search box doesn't perform actual filtering. The actual filtering is performed when you press 'Apply' button.

![изображение](https://user-images.githubusercontent.com/2705882/54186338-da12a300-44b3-11e9-8554-52e727340dbb.png)

**Comment by corganfuzz** [comment]
@bkulov sorry about this late reply I've been busy. but here it goes:

Let's talk about **_quickfilter_** first.

When you click on the filter chip you get a search box that says "enter filter value" as far as I know this is bound to the **onFilterringDone()** method.

This search box is able to do 2 things for me:

1. It's able to filter the data values my column it is on, by typing a letter.
2. It's able to return data from the grid based on the selection of (1), if you attached the **onfilteringDone()** method to the `<igx-grid>` tag 

Therefore, in (2) I can do a lot of things with this data such as: pass it to a chart or to a kpi card , basically  passing this data to a sibling component. 

So to ask what is the benefit of this is like saying what is the benefit of passing data to sibling components based on the user filtering selection which will be **instant** data binding.  I get changes on my UI as I type, which is pretty appealing for the user. (they had expressively said so)

Now, let's see **_excelfilterstyle_**:

You have the same Search box, but now you've introduced 2 new buttons (**apply** and **cancel**)

 1. if you have the **same** search button in it, then:

      a. Why change the behavior this search box that already exists on the previous filtering method?

      b. Why making it different. It's the same search box as above. 

      c.  Why is **onfilteringDone()** not attached to it ? I know it filters the values of that column but 

      d.  Why take off the other functionality it was already proving before with **onFilteringDone()** 
           method, or the ability to bind it to it.

 2. if you introduced 2 new buttons on the mix, wouldn't it make sense to expose 2 new methods to these two new buttons. at the end a button end goal is to emit an event when is clicked. It's its sole purpose

In conclusion, I think quickfilter and excelfilter shouldn't be THAT different. the only difference between them should be the fact that now you are able to see your data column within checkboxes and that's it. 

the current state of the component makes me think somebody forgot to expose the onFilteringDone() method in the search box, that filters these checkboxes.

**Comment by kdinev** [comment]
@corganfuzz You bring up a good point that the two filtering modes may have different event/event sequences. We will investigate, note the differences and make it consistent. Thanks for bringing this up and we will keep you posted in the issue!
